### ISSUE TITLE
separator argument ImageClassificationData.from_data_frame for multi label dataset

### ISSUE THREAD

**Issue by aniketmaurya** [PyTorchLightning/lightning-flash#728]
## 🚀 Feature

for [plant pathology](https://www.kaggle.com/c/plant-pathology-2021-fgvc8) kaggle dataset, after loading the csv/df with `ImageClassificationData.from_data_frame` I am getting dm.num_classes 11 but it should be 6.

There should be a way to pass how labels are separated in the df/csv - by space, comma or any other character.

**Comment by ethanwharris** [comment]
Hi @aniketmaurya great suggestion! It's worth noting that we don't currently support this format at all. For multi-label, our CSV loading code assumes that each additional label is in a new column rather than a delimited string in a single column. Support for this, along with the ability to specify the delimiter, would be much appreciated. Is this something you would be willing to work on? The main class to be updated is here: https://github.com/PyTorchLightning/lightning-flash/blob/8f306d49b24aed09d37d083883f6c5636b26a45e/flash/core/data/data_source.py#L504

**Comment by aniketmaurya** [comment]
Sure, Ethan! I will start working on this. thank you for pointing the class to be updated :D

**Comment by aniketmaurya** [comment]
hey @ethanwharris, is it okay to use sklearn's MultiLabelBinarizer for converting labels into sparse encoding? or should I keep minimum dependency and write a custom logic here?

**Comment by ethanwharris** [comment]
Still relevant, possible duplicate of #836

**Comment by ethanwharris** [comment]
Closing this as plant pathology can now be loaded succesfully. We don't support passing a seperator but it will be inferred and can be either a space or a comma :smiley:
